src/helpers/perfMaps.ts

```diff
@@ -0,0 +1,35 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as core from "@actions/core";
+import {promisify} from "util";
+
+const copyFile = promisify(fs.copyFile);
+const readdir = promisify(fs.readdir);
+const exists = promisify(fs.exists);
+
+export async function harvestPerfMaps(profileFolder: string): Promise<void> {
+  const outFiles = fs
+    .readdirSync(profileFolder)
```

Comment:
Use `readDir` from promises instead of the sync one, we will always be better off using those: https://advancedweb.hu/do-not-use-fs-sync-methods-in-javascript-use-fs-promises-instead/